cert.go

```diff
@@ -438,12 +458,14 @@ func (opts *CertificateOptions) CreateCertificate(wd depot.Depot) error {
 // it expires within the duration `after` and creates a new certificate if
 // either condition is met. True is returned if a certificate is created,
 // false otherwise. If the certificate is a CA, the behavior is undefined.
-func (opts *CertificateOptions) CreateCertificateOnExpiration(wd depot.Depot, after time.Duration) (bool, error) {
+func (opts *CertificateOptions) CreateCertificateOnExpiration(wd Depot, after time.Duration) (bool, error) {
 	dne := true
 	var created bool
 	var err error
 
-	if depot.CheckCertificate(wd, opts.CommonName) {
+	if exists, err := CheckCertificateWithError(wd, opts.CommonName); err != nil {
+		return created, err
```

Comment:
Wrap error. Would it make more sense to just return false here, since created hasn't even been set yet?

Reply:
I would rather propagate up the error since if something is going on with the database it is useful for cedar to know.

Reply:
My point was that it's not worth it to declare a variable early in the function and return it when you haven't even used yet, but I don't care that much.